code/datums/status_effects/status_effect.dm

```diff
@@ -112,16 +112,16 @@
 	. = FALSE
 	var/datum/status_effect/S1 = effect
 	LAZYINITLIST(status_effects)
+	var/list/arguments = args.Copy()
 	for(var/datum/status_effect/S in status_effects)
 		if(S.id == initial(S1.id) && S.status_type)
 			if(S.status_type == STATUS_EFFECT_REPLACE)
 				S.be_replaced()
 			else if(S.status_type == STATUS_EFFECT_REFRESH)
-				S.refresh()
+				S.refresh(arglist(arguments))
 				return
 			else
 				return
-	var/list/arguments = args.Copy()
 	arguments[1] = src
 	S1 = new effect(arguments)
```

Comment:
Should refresh() get the  same arguments as creating a new effect? ditto for be_replaced()

Reply:
If it doesn't then the refresh() proc of the ant debuff doesn't function, meaning that rather than adding more ants onto someone, it just deletes all the existing ants and then restarts with the new ants. Basically, someone could remove like 40 ants by just putting a single ant on themselves.

Reply:
why are we passing in arguments to refresh() before setting the first item in the list to src then? shouldn't they have the same behavior?

Reply:
I didn't realize they didn't have the same behavior. Trying to get the refresh() proc to get args in the first place was confusing in the first place, and the last thing I want to do is fuck up some actually important code like this. Do you know of some other way I'd be able to move the args I need into the proc, or a cleaner way of doing this refresh() one?

Reply:
you should move the `arguments[1] = src` up to just under `var/list/arguments = args.Copy()` I'd think

Reply:
Done.